File: platform/platform-grid/src/test/java/com/yazino/platform/persistence/PrimaryKeyLoader.java
package com.yazino.platform.persistence;

import org.springframework.dao.DataAccessException;
import org.springframework.jdbc.core.PreparedStatementCallback;

import java.math.BigDecimal;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class PrimaryKeyLoader implements PreparedStatementCallback<BigDecimal> {
    @Override
    public BigDecimal doInPreparedStatement(final PreparedStatement statement)
            throws SQLException, DataAccessException {
        statement.execute();

        ResultSet rs = null;
        try {
            rs = statement.getGeneratedKeys();
            if (rs.next()) {
                return rs.getBigDecimal(1);
            }
            return null;

        } finally {
            if (rs != null) {
                try {
                    rs.close();

                } catch (SQLException e) {
                    // ignored
                }
            }
        }
    }
}

File: platform/control-centre/src/main/scala/strata/server/lobby/controlcentre/repository/PrimaryKeyLoader.scala
package strata.server.lobby.controlcentre.repository

import java.lang.{Long => JavaLong}
import org.apache.commons.logging.LogFactory
import org.springframework.jdbc.core.PreparedStatementCallback
import java.sql.PreparedStatement
import java.sql.ResultSet
import java.sql.SQLException

class PrimaryKeyLoader extends PreparedStatementCallback[JavaLong] {
    private val LOG = LogFactory.getLog(classOf[PrimaryKeyLoader])

    def doInPreparedStatement(statement: PreparedStatement): JavaLong = {
        statement.execute
        var rs: ResultSet = null
        try {
            rs = statement.getGeneratedKeys
            if (rs.next) {
                val key: JavaLong = rs.getLong(1)
                LOG.debug("readAggregate key " + key)
                key
            } else {
                null
            }
        }
        finally {
            if (rs != null) {
                try {
                    rs.close()
                }
                catch {
                    case e: SQLException => {
                    }
                }
            }
        }
    }
}
